src/app/services/auth-guard.service.ts

```diff
@@ -1,15 +1,22 @@
 import { Injectable } from '@angular/core';
 import { Router, CanActivate } from '@angular/router';
 import { AuthService } from './auth.service';
-import { Observable } from 'rxjs';
+import { Store, select } from '@ngrx/store';
+import { State } from '../reducers';
 
 @Injectable({
   providedIn: 'root'
 })
 export class AuthGuardService implements CanActivate {
-  private isLoggedIn: Observable<boolean>;
-  constructor(public auth: AuthService, public router: Router) {
-    this.isLoggedIn = this.auth.isAuthentificated();
+  private isLoggedIn: boolean;
+  constructor(
+    public auth: AuthService,
+    public router: Router,
+    private store: Store<State>
+  ) {
+    this.store.pipe(select('auth', 'isAuthenticated')).subscribe((value) => {
```

Comment:
The value in the store does not respect values in Local Storage so every page refresh leads to the login page.